embedded/sql/engine_test.go

```diff
@@ -388,8 +391,179 @@ func TestAddColumn(t *testing.T) {
+	require.NoError(t, err)
+
+	res, err := engine.Query("SELECT id, name, surname FROM table1", nil, nil)
+	require.NoError(t, err)
+
+	row, err := res.Read()
+	require.NoError(t, err)
+
+	require.EqualValues(t, 1, row.ValuesByPosition[0].Value())
+	require.EqualValues(t, "John", row.ValuesByPosition[1].Value())
+	require.EqualValues(t, "Smith", row.ValuesByPosition[2].Value())
+
+	_, err = res.Read()
+	require.ErrorIs(t, err, ErrNoMoreRows)
+
+	res.Close()
+	st.Close()
+
+	// Reopen store
+	st, err = store.Open("sqldata_add_column", store.DefaultOptions())
+	require.NoError(t, err)
+
+	engine, err = NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	err = engine.SetCurrentDatabase("db1")
+	require.NoError(t, err)
+
+	res, err = engine.Query("SELECT id, name, surname FROM table1", nil, nil)
+	require.NoError(t, err)
+
+	row, err = res.Read()
+	require.NoError(t, err)
+
+	require.EqualValues(t, 1, row.ValuesByPosition[0].Value())
+	require.EqualValues(t, "John", row.ValuesByPosition[1].Value())
+	require.EqualValues(t, "Smith", row.ValuesByPosition[2].Value())
+
+	_, err = res.Read()
+	require.ErrorIs(t, err, ErrNoMoreRows)
+
+	res.Close()
+	st.Close()
+}
+
+func TestRenameColumn(t *testing.T) {
+	st, err := store.Open("sqldata_rename_column", store.DefaultOptions())
+	require.NoError(t, err)
+	defer os.RemoveAll("sqldata_rename_column")
+
+	engine, err := NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN name TO surname", nil, nil)
+	require.ErrorIs(t, err, ErrNoDatabaseSelected)
+
+	_, _, err = engine.Exec("CREATE DATABASE db1", nil, nil)
+	require.NoError(t, err)
+
+	err = engine.SetCurrentDatabase("db1")
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("CREATE TABLE table1 (id INTEGER AUTO_INCREMENT, name VARCHAR[50], PRIMARY KEY id)", nil, nil)
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("CREATE INDEX ON table1(name)", nil, nil)
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("INSERT INTO table1(name) VALUES('John'), ('Sylvia'), ('Robocop') ", nil, nil)
+	require.NoError(t, err)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN name TO name", nil, nil)
+	require.ErrorIs(t, err, ErrSameOldAndNewColumnName)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN name TO id", nil, nil)
+	require.ErrorIs(t, err, ErrColumnAlreadyExists)
+
+	_, _, err = engine.Exec("ALTER TABLE table2 RENAME COLUMN name TO surname", nil, nil)
+	require.ErrorIs(t, err, ErrTableDoesNotExist)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN surname TO name", nil, nil)
+	require.ErrorIs(t, err, ErrColumnDoesNotExist)
+
+	_, _, err = engine.Exec("ALTER TABLE table1 RENAME COLUMN name TO surname", nil, nil)
+	require.NoError(t, err)
+
+	res, err := engine.Query("SELECT id, surname FROM table1 ORDER BY surname", nil, nil)
+	require.NoError(t, err)
+	defer res.Close()
```

Comment:
deferring the close may silent unexpected errors, the data folder will be physically deleted anyways, so if the close call is not reached, it shouldn't be an issue